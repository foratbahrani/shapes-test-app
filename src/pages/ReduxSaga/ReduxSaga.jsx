import React, { Component } from 'react';
import { connect } from 'react-redux';
import GridTitleLabel from '../../components/GridTitleLabel/GridTitleLabel';
import ItemsGridBox from '../../components/ItemsGridBox/ItemsGridBox';
import Toolbar from '../../components/Toolbar/Toolbar';
import Container from '../../containers/Container/Container';
import { fetchShapesStart } from '../../store/shapes/shapes.actions';

export class ReduxSaga extends Component {
	componentDidMount() {
		const { fetchShapes } = this.props;
		if (fetchShapes) fetchShapes();
	}

	render() {
		return (
			<Container size='lg'>
				<Toolbar />
				<GridTitleLabel />
				<ItemsGridBox />
			</Container>
		);
	}
}

const mapDispatchToProps = {
	fetchShapes: fetchShapesStart,
};

export default connect(null, mapDispatchToProps)(ReduxSaga);
