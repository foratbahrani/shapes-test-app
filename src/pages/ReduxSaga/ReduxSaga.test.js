import { shallow } from 'enzyme';
import React from 'react';
import { ReduxSaga } from './ReduxSaga';

it('matches snapshot', () => {
	expect(shallow(<ReduxSaga fetchShapes={() => {}} />)).toMatchSnapshot();
});

it('calls fetchShapes', () => {
	const fetchShapes = jest.fn().mockReturnValue('');
	shallow(<ReduxSaga fetchShapes={fetchShapes} />);
	expect(fetchShapes.mock.calls.length).toBe(1);
});
