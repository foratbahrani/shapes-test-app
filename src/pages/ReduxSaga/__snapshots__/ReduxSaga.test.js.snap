// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Container
  size="lg"
>
  <Toolbar />
  <Connect(GridTitleLabel) />
  <Connect(HOC) />
</Container>
`;
